File: android/talk-talk/app/src/main/java/com/grabieckacper/talktalk/common/TalkType.kt
package com.grabieckacper.talktalk.common

import com.grabieckacper.talktalk.R

enum class TalkType(val stringResource: StringValue.StringResource) {
    ALL(stringResource = StringValue.StringResource(resId = R.string.talk_type_all)),
    UNREAD(stringResource = StringValue.StringResource(resId = R.string.talk_type_unread)),
    GROUP(stringResource = StringValue.StringResource(resId = R.string.talk_type_group))
}

File: ios/TalkTalk/TalkTalk/Common/TalkType.swift
//
//  TalkType.swift
//  TalkTalk
//
//  Created by Kacper Grabiec on 05/09/2024.
//

import Foundation

enum TalkType : String, CaseIterable {
    case all = "All"
    case unread = "Unread"
    case group = "Groups"
    
    func localizedString() -> String {
        return NSLocalizedString(self.rawValue, comment: "")
    }
}
